indico/modules/events/registration/client/js/form/fields/MultiChoiceInput.jsx

```diff
@@ -0,0 +1,105 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2021 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {Field} from 'react-final-form';
+
+import {FinalCheckbox, FinalField} from 'indico/react/forms';
+import {Translate} from 'indico/react/i18n';
+
+import {Choices, choiceShape} from './ChoicesSetup';
+
+import '../../../styles/regform.module.scss';
+
+export default function MultiChoiceInput({htmlName, disabled, choices, withExtraSlots}) {
+  // TODO: billable/price
+  // TODO: places left
+  // TODO: set value as `{uuid: places}` (when adding final-form integration)
+  // TODO: disable options triggering price changes after payment (or warn for managers)
+  // TODO: warnings for deleted/modified choices
+  const [value, setValue] = useState({});
+
+  const makeHandleChange = choice => evt => {
+    setValue(prev => ({...prev, [choice.id]: evt.target.checked ? 1 : 0}));
```

Comment:
The suggestion makes sense, but I don't think we can merge the two since one uses checked for the choice and the other one uses value for the number of slots